SwrveSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -1059,22 +1057,22 @@ protected boolean downloadAssetSynchronously(final String assetPath) {
     protected void saveCampaignSettings() {
         try {
             // Save campaign state
-            JSONObject jsonSettings = new JSONObject();
+            JSONObject campaignStateJson = new JSONObject();
             Iterator<SwrveBaseCampaign> itCampaign = campaigns.iterator();
```

Comment:
Yes, the state has to be stored even though we might not have that campaign, it is the whole purpose of this change